### PR TITLE
Support background audio

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Release Notes

- **New Features**
  - Introduced `Audio` enum value to support audio formats (ogg, wav, mp3).
  - Added `AudioClip`, `AudioSource`, and `AudioManager` classes for enhanced audio management and playback.
  - Implemented `AudioLoader` for loading audio assets seamlessly.
  - Added support for automatic playback when the `AudioSource` component is enabled.
  - Expanded core module exports to include new audio-related classes and types.

- **Bug Fixes**
  - Corrected punctuation in the `SourceFont` enum comment.

- **Tests**
  - Added unit tests for `AudioSource` to ensure proper functionality and playback behavior.
  - Introduced a base64-encoded audio constant for testing purposes.

- **Documentation**
  - Enhanced `request` function documentation to clarify its purpose and parameters.

### CODE REVIEW

File: packages/core/src/audio/AudioSource.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
+import { deepClone, ignoreClone } from "../clone/CloneManager";
+import { Component } from "../Component";
+import { Entity } from "../Entity";
+import { AudioClip } from "./AudioClip";
+import { AudioManager } from "./AudioManager";
+
+/**
+ * Audio Source Component.
+ */
+export class AudioSource extends Component {
```
Comment (singlecoder) [inline packages/core/src/audio/AudioSource.ts:10]: AudioSource looks more like an asset rather than a component

Comment (GuoLei1990) [reply packages/core/src/audio/AudioSource.ts:10]: `AudioSource` and `AudioListener` is OK! Source here is the source of the sound

File: packages/core/src/audio/AudioManager.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  private static _gainNode: GainNode;
+  private static _isResuming = false;
+
+  static getContext(): AudioContext {
+    let context = AudioManager._context;
+    if (!context) {
+      AudioManager._context = context = new window.AudioContext();
+
+      // Safari can't resume audio context without element interaction
+      const document = window.document;
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/audio/AudioManager.ts]: Done

**Review by feyabibi — APPROVED** [review]
The code has been reviewed and looks good

**Pull request merged**